File: HomeWork/HomeWork7/HomeWork7.java
package HomeWork.HomeWork7;

import java.util.*;

public class HomeWork7{
    public static int minMeetingRooms(int[][] intervals) {
        HashMap<Integer, Integer> map = new HashMap<Integer,Integer>(); // Key --> Minute Value --> meetings at minute
        for(int i = 0; i < intervals.length; i++){
            for(int x = intervals[i][0]; x < intervals[i][1]; x++){
                map.put(x, map.getOrDefault(x,0) + 1);
            }
        }
        return Collections.max(map.values());
    }
    public static void main(String[] args) {
        int[][] mat = {{0,30},{5,10},{15,20}};
        System.out.println(minMeetingRooms(mat));
    }

}

File: HomeWork/HomeWork7/HomeWork7.py
def minMeetingRooms(intervals):
    meeting_map = {}
    
    for interval in intervals:
        start, end = interval
        
        for minute in range(start, end):
            meeting_map[minute] = meeting_map.get(minute, 0) + 1
    
    return max(meeting_map.values())

if __name__ == "__main__":
    intervals = [[0, 30], [5, 10], [15, 20]]
    print(minMeetingRooms(intervals))
